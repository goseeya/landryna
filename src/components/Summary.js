const Summary = ({ summaryData }) => {
  return (
    <div className="Summary">
      <p>
        Average population of all the countries:{" "}
        {Math.round((summaryData.avgPop / 1000000) * 10) / 10}
      </p>
      <p>Country with the smallest area: {summaryData.smallestCountry}</p>
      <p>Country with the biggest area: {summaryData.biggestCountry}</p>
    </div>
  );
};

export default Summary;
