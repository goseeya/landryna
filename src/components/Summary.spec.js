import React from "react";
import { create } from "react-test-renderer";
import { shallow } from "enzyme";
import Summary from "./Summary";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

const mockedSummary = {
  avgPop: 3455355,
  smallestCountry: "Cyprus",
  biggestCountry: "Poland",
};

const createFromProps = (props) => {
  <Summary summaryData={props} />;
};

describe("Summary", () => {
  describe("Snapshot", () => {
    it("should match base snapshot", () => {
      const component = create(createFromProps(mockedSummary));
      expect(component.toJSON()).toMatchSnapshot();
    });
    it("should render summary row", () => {
      const component = shallow(<Summary summaryData={mockedSummary} />);
      expect(component.find(".summary")).toHaveLength(1);
    });
  });
});
