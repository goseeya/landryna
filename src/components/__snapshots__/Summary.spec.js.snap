// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary Snapshot should match base snapshot 1`] = `null`;
